accounts-db/src/accounts_hash.rs

```diff
@@ -87,21 +87,54 @@ impl AccountHashesFile {
         if self.writer.is_none() {
             // we have hashes to write but no file yet, so create a file that will auto-delete on drop
 
-            let mut data = tempfile_in(&self.dir_for_temp_cache_files).unwrap_or_else(|err| {
-                panic!(
-                    "Unable to create file within {}: {err}",
-                    self.dir_for_temp_cache_files.display()
-                )
-            });
+            let get_file = || -> Result<_, std::io::Error> {
+                let mut data = tempfile_in(&self.dir_for_temp_cache_files).unwrap_or_else(|err| {
+                    panic!(
+                        "Unable to create file within {}: {err}",
+                        self.dir_for_temp_cache_files.display()
+                    )
+                });
+
+                // Theoretical performance optimization: write a zero to the end of
+                // the file so that we won't have to resize it later, which may be
+                // expensive.
+                assert!(self.capacity > 0);
+                data.seek(SeekFrom::Start((self.capacity - 1) as u64))?;
+                data.write_all(&[0])?;
+                data.rewind()?;
+                data.flush()?;
+                Ok(data)
+            };
 
-            // Theoretical performance optimization: write a zero to the end of
-            // the file so that we won't have to resize it later, which may be
-            // expensive.
-            data.seek(SeekFrom::Start((self.capacity - 1) as u64))
-                .unwrap();
-            data.write_all(&[0]).unwrap();
-            data.rewind().unwrap();
-            data.flush().unwrap();
+            // Retry 5 times for allocation the AccountHashFile. The memory maybe fragmented and
+            // causes memory allocation failure. Therefore, retry after failure. Hoping that the
+            // kernel can defrag the memory and allocation retries can succeed.
+            let mut num_retries = 0;
+            let data = loop {
+                num_retries += 1;
+
+                match get_file() {
```

Comment:
Perhaps introducing some delay between retries is desirable (maybe even linearly increasing too) ?

Reply:
yeah. good idea.